### ISSUE TITLE
AWS IoT Integration Creating TLS Factory Failed

### ISSUE THREAD

**Issue by xhr0428** [thingsboard/thingsboard#2842]
**Component**

 * Generic

**Description**
I'm testing AWS Integration with cloud.thingsboard.io following your video tutorial. But I keep receiving creating TLS factory failed error after uploading AWS rootCA file. I have tried both  VeriSign Class 3 Public Primary G5 root CA certificate and  Amazon Root CA 1. Can anybody help me with it? Thanks

java.lang.RuntimeException: Creating TLS factory failed!
	at org.thingsboard.integration.mqtt.credentials.CertPemClientCredentials.initSslContext(CertPemClientCredentials.java:93)
	at org.thingsboard.integration.mqtt.AbstractMqttIntegration.initSslContext(AbstractMqttIntegration.java:196)
	at org.thingsboard.integration.mqtt.AbstractMqttIntegration.initClient(AbstractMqttIntegration.java:164)
	at org.thingsboard.integration.mqtt.basic.BasicMqttIntegration.init(BasicMqttIntegration.java:74)
	at org.thingsboard.integration.mqtt.aws.AwsIotIntegration.init(AwsIotIntegration.java:46)
	at org.thingsboard.integration.mqtt.AbstractMqttIntegration.update(AbstractMqttIntegration.java:99)
	at org.thingsboard.server.service.integration.DefaultPlatformIntegrationService.lambda$updateIntegration$2(DefaultPlatformIntegrationService.java:367)
	at com.google.common.util.concurrent.TrustedListenableFutureTask$TrustedFutureInterruptibleTask.runInterruptibly(TrustedListenableFutureTask.java:125)
	at com.google.common.util.concurrent.InterruptibleTask.run(InterruptibleTask.java:69)
	at com.google.common.util.concurrent.TrustedListenableFutureTask.run(TrustedListenableFutureTask.java:78)
	at java.util.concurrent.ForkJoinTask$RunnableExecuteAction.exec(ForkJoinTask.java:1402)
	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289)
	at java.util.concurrent.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1056)
	at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1692)
	at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:157)
Caused by: java.io.IOException: overrun, bytes = 924
	at javax.crypto.EncryptedPrivateKeyInfo.<init>(EncryptedPrivateKeyInfo.java:92)
	at org.thingsboard.integration.mqtt.credentials.CertPemClientCredentials.getKeySpec(CertPemClientCredentials.java:186)
	at org.thingsboard.integration.mqtt.credentials.CertPemClientCredentials.readPrivateKeyFile(CertPemClientCredentials.java:173)
	at org.thingsboard.integration.mqtt.credentials.CertPemClientCredentials.createAndInitKeyManagerFactory(CertPemClientCredentials.java:104)
	at org.thingsboard.integration.mqtt.credentials.CertPemClientCredentials.initSslContext(CertPemClientCredentials.java:87)
	... 14 more

**Environment**
using cloud.thingsboard.io

**Comment by ashvayka** [comment]
Looks like the PEM file you provide is not correct:

org.thingsboard.integration.mqtt.credentials.CertPemClientCredentials.getKeySpec(CertPemClientCredentials.java:186)

**Comment by xhr0428** [comment]
> Looks like the PEM file you provide is not correct:
> 
> org.thingsboard.integration.mqtt.credentials.CertPemClientCredentials.getKeySpec(CertPemClientCredentials.java:186)

Hi, I followed the video instruction, created a new set of certificates, downloaded cert file and private key, downloaded Amazon Root CA 1 and attached the following policy
{
  "Version": "2012-10-17",
  "Statement": [
    {
      "Effect": "Allow",
      "Action": "iot:*",
      "Resource": "*"
    }
  ]
}
I could post message to AWS IoT using mosquitto_pub and the crt, key, pem files I downloaded. While uploading the same set of files to thingsboard, I saw this error. 

What would be the correct pem file to use? Thanks
